src/psyclone/psyir/frontend/fparser2.py

```diff
@@ -3370,14 +3358,20 @@ def _do_construct_handler(self, node, parent):
             raise NotImplementedError("Unsupported Loop")
 
         # Process loop body (ignore 'do' and 'end do' statements)
-        # Keep track of the comments before the 'do' statement
+        # Keep track of the comments and directives before the 'do' statement
         loop_body_nodes = []
         preceding_comments = []
         found_do_stmt = False
         for child in node.content:
             if isinstance(child, Fortran2003.Comment) and not found_do_stmt:
                 self.process_comment(child, preceding_comments)
                 continue
+            if isinstance(child, Fortran2003.Directive) and not found_do_stmt:
+                directive = self._directive_handler(child, None)
```

Comment:
Just to see if I understand this correctly:
- the _directive_handler can be called from the self.handlers map when we find a directive in an execution regions (presumably a standalone directive),
- but it also has to be called from inside dos and ifs because region directives are inside the fparser node, not beside it, e.g.: `If_Construct(Directive('!$omp target'), , If_Then_Stmt(Level_4_Expr(Name('i'), `
?

Reply:
Yeah - fparser puts them wherever it would have put the comment before, so at the end of declarations, in the main block of an execution region, or attached to a construct.